roiextractors/extractors/caiman/caimansegmentationextractor.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import h5py
+from lazy_ops import DatasetView
+from ...segmentationextractor import SegmentationExtractor
+from ...extraction_tools import _pixel_mask_extractor
+
+class CaimanSegmentationExtractor(SegmentationExtractor):
+    """
+    This class inherits from the SegmentationExtractor class, having all
+    its funtionality specifically applied to the dataset output from
+    the \'CNMF-E\' ROI segmentation method.
+    """
+    extractor_name = 'CaimanSegmentation'
+    installed = True  # check at class level if installed or not
+    is_writable = False
+    mode = 'file'
+    installation_mesg = ""  # error message when not installed
+
+    def __init__(self, filepath):
+        """
+        Parameters
+        ----------
+        filepath: str
+            The location of the folder containing caiman *.hdmf output file.
+        """
+        SegmentationExtractor.__init__(self)
+        self.filepath = filepath
+        self._dataset_file = self._file_extractor_read()
+        self._roi_response = self._trace_extractor_read('F_dff')
+        self._masks, self._mask_roi_ids, self._mask_ids = self._image_mask_sparse_read()
+        self._roi_response_dict = {'Fluorescence': self._roi_response,
+                                   'Neuropil':self._trace_extractor_read('C'),
+                                   'Deconvolved':self._trace_extractor_read('S')}
+
+        self._raw_movie_file_location = self._dataset_file['params']['data']['fnames'][0].decode('utf-8')
+        self._sampling_frequency = self._dataset_file['params']['data']['fr'].value
+        # file close:
+        # self._file_close()
+
+    def _file_close(self):
```

Comment:
See use of `__del__` e.g. https://github.com/catalystneuro/roiextractors/blob/4e5117ecabcb2340ca72166e32db96d411add351/roiextractors/extractors/bioformatsimagingextractors/bioformatsimagingbaseextractor.py#L26-L28

Reply:
fixed 007150d079ad5a8c25664a4c186db9f14d7c6821